plugins/modules/docker_plugin.py

```diff
@@ -0,0 +1,343 @@
+            self.client.fail(text_type(e))
+
+        if plugin is None:
+            return None
+        else:
+            return plugin
+
+    def has_different_config(self):
+        """
+        Return the list of differences between the current parameters and the existing plugin.
+
+        :return: list of options that differ
+        """
+        differences = DifferenceTracker()
+        if self.parameters.plugin_options:
+            if not self.existing_plugin.settings:
+                differences.add('plugin_options', parameters=self.parameters.plugin_options, active=self.existing_plugin.settings['Env'])
+            else:
+                existing_options_list = self.existing_plugin.settings['Env']
+                existing_options = parse_options(existing_options_list)
+
+                for key, value in self.parameters.plugin_options.items():
+                    options_count = 0
+                    if ((not existing_options.get(key) and value) or
+                            not value or
+                            value != existing_options[key]):
+                        differences.add('plugin_options.%s' % key,
+                                        parameter=value,
+                                        active=self.existing_plugin.settings['Env'][options_count])
+
+        return differences
+
+    def install_plugin(self):
+        if not self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin = self.dclient.plugins.install(self.parameters.plugin_name, None)
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Installed plugin %s" % self.parameters.plugin_name)
+            self.results['changed'] = True
+
+    def remove_plugin(self):
+        force = self.parameters.force_remove
+        if self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin.remove(force)
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Removed plugin %s" % self.parameters.plugin_name)
+            self.results['changed'] = True
+
+    def update_plugin(self):
+        if self.existing_plugin:
+            differences = self.has_different_config()
+                if not differences.empty:
+                    if not self.check_mode:
+                        try:
+                            self.existing_plugin.configure(prepare_options(self.parameters.plugin_options))
+                        except APIError as e:
+                            self.client.fail(text_type(e))
+                    self.results['actions'].append("Updated plugin %s settings" % self.parameters.plugin_name)
+                    self.results['changed'] = True
+            else:
+                self.fail("Cannot update the plugin: Plugin does not exist")
```

Comment:
This block needs to be dedented by 4 spaces:
```suggestion
            if not differences.empty:
                if not self.check_mode:
                    try:
                        self.existing_plugin.configure(prepare_options(self.parameters.plugin_options))
                    except APIError as e:
                        self.client.fail(text_type(e))
                self.results['actions'].append("Updated plugin %s settings" % self.parameters.plugin_name)
                self.results['changed'] = True
        else:
            self.fail("Cannot update the plugin: Plugin does not exist")
```